core/integration/module_test.go

```diff
@@ -1890,6 +1895,92 @@ class Test {
 	}
 }
 
+func (ModuleSuite) TestCodegenOptionals(ctx context.Context, t *testctx.T) {
+	// Same code as TestOptionalDefaults since it guarantees this is being
+	// registered correctly and equally by all SDKs.
+	src := `package main
+
+import "fmt"
+
+type Dep struct {}
+
+func (m *Dep) Ctl(
+	a string,
+	// +optional
+	b *string,
+	// +default="foo"
+	c string,
```

Comment:
So to check my understanding:
- With just `optional`, this would be `String` instead of `String!` (which marks it as required in graphql)
- With just `default`, this would be `String! = "foo"`
- With both, we get `String = "foo"`

So with this change, *all* of the above are interpreted as "optional" for go? That makes sense to me, it brings things more inline with the other SDKs, and go doesn't realllly have either concept, so it makes sense to blur them a bit imo.

Reply:
> So with this change, _all_ of the above are interpreted as "optional" for go? 

Yes, that's correct. 

`!` doesn't mean "required", it just means "doesn't allow `null`". An optional value in GraphQL is one that either has a default value or is nullable. That's because GraphQL implicitly adds `null` as the default value for any nullable without a default.